src/validator/MaintainValidator.cpp

```diff
@@ -16,6 +17,51 @@
 
 namespace nebula {
 namespace graph {
+
+Status SchemaValidator::validateColumns(const std::vector<ColumnSpecification*> &columnSpecs,
+                                        meta::cpp2::Schema &schema) {
+    auto status = Status::OK();
+    std::unordered_set<std::string> nameSet;
+    for (auto& spec : columnSpecs) {
+        if (nameSet.find(*spec->name()) != nameSet.end()) {
+            return Status::SemanticError("Duplicate column name `%s'", spec->name()->c_str());
+        }
+        nameSet.emplace(*spec->name());
+        meta::cpp2::ColumnDef column;
+        auto type = spec->type();
+        column.set_name(*spec->name());
+        column.set_type(type);
+        column.set_nullable(spec->isNull());
+        if (meta::cpp2::PropertyType::FIXED_STRING == type) {
+            column.set_type_length(spec->typeLen());
+        }
+
+        if (spec->hasDefaultValue()) {
+            if (!evaluableExpr(spec->getDefaultValue())) {
+                return Status::SemanticError("Wrong default value experssion `%s'",
+                                              spec->getDefaultValue()->toString().c_str());
+            }
+            QueryExpressionContext ctx(nullptr, nullptr);
+            auto defaultValueExpr = spec->getDefaultValue();
+            auto& value = defaultValueExpr->eval(ctx);
+            auto valStatus = SchemaUtil::toSchemaValue(type, value);
+            if (!valStatus.ok()) {
+                return valStatus.status();
+            }
+            // When the timestamp value is string, need to save the int value
+            if (type == meta::cpp2::PropertyType::TIMESTAMP && value.isStr()) {
+                ConstantExpression newExpr(std::move(valStatus).value());
+                column.set_default_value(newExpr.encode());
```

Comment:
If you store the ConstantExpression it will produce the default value when insert new data.